src/gs-window-x11.c

```diff
@@ -1865,368 +1532,244 @@ gs_window_real_motion_notify_event (GtkWidget      *widget,
-					 g_param_spec_string ("keyboard-command",
-							      NULL,
-							      NULL,
-							      NULL,
-							      G_PARAM_READWRITE));
-
-	g_object_class_install_property (object_class,
-					 PROP_MONITOR,
-					 g_param_spec_int ("monitor",
-							   "Xinerama monitor",
-							   "The monitor (in terms of Xinerama) which the window is on",
-							   0,
-							   G_MAXINT,
-							   0,
-							   G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
-
-}
-
-static void
-create_info_bar (GSWindow *window)
-{
-	window->priv->info_bar = gtk_info_bar_new ();
-	gtk_widget_set_no_show_all (window->priv->info_bar, TRUE);
-	gtk_box_pack_end (GTK_BOX (window->priv->vbox), window->priv->info_bar, FALSE, FALSE, 0);
-}
-
-static gboolean
-on_panel_draw (GtkWidget    *widget,
-	       cairo_t      *cr,
-	       GSWindow     *window)
-{
-	cairo_set_source_rgba (cr, 0.0, 0.0, 0.0, 1.0);
-	cairo_paint (cr);
+	g_type_class_add_private(klass, sizeof(GSWindowPrivate));
+
+	signals[ACTIVITY] = g_signal_new("activity", G_TYPE_FROM_CLASS(object_class), G_SIGNAL_RUN_LAST,
+		G_STRUCT_OFFSET(GSWindowClass, activity), NULL, NULL, gs_marshal_BOOLEAN__VOID, G_TYPE_BOOLEAN, 0);
+	signals[DEACTIVATED] = g_signal_new("deactivated", G_TYPE_FROM_CLASS(object_class), G_SIGNAL_RUN_LAST,
+		G_STRUCT_OFFSET(GSWindowClass, deactivated), NULL, NULL, g_cclosure_marshal_VOID__VOID, G_TYPE_NONE, 0);
+
+	g_object_class_install_property(
+		object_class, PROP_OBSCURED, g_param_spec_boolean("obscured", NULL, NULL, FALSE, G_PARAM_READABLE));
+	g_object_class_install_property(
+		object_class, PROP_DIALOG_UP, g_param_spec_boolean("dialog-up", NULL, NULL, FALSE, G_PARAM_READABLE));
+	g_object_class_install_property(
+		object_class, PROP_LOCK_ENABLED, g_param_spec_boolean("lock-enabled", NULL, NULL, FALSE, G_PARAM_READWRITE));
+	g_object_class_install_property(
+		object_class, PROP_LOGOUT_ENABLED, g_param_spec_boolean("logout-enabled", NULL, NULL, FALSE, G_PARAM_READWRITE));
+	g_object_class_install_property(
+		object_class, PROP_LOGOUT_TIMEOUT, g_param_spec_long("logout-timeout", NULL, NULL, -1, G_MAXLONG, 0, G_PARAM_READWRITE));
+	g_object_class_install_property(
+		object_class, PROP_LOGOUT_COMMAND, g_param_spec_string("logout-command", NULL, NULL, NULL, G_PARAM_READWRITE));
+	g_object_class_install_property(
+		object_class, PROP_STATUS_MESSAGE, g_param_spec_string("status-message", NULL, NULL, NULL, G_PARAM_READWRITE));
+	g_object_class_install_property(
+		object_class, PROP_KEYBOARD_ENABLED, g_param_spec_boolean("keyboard-enabled", NULL, NULL, FALSE, G_PARAM_READWRITE));
+	g_object_class_install_property(
+		object_class, PROP_KEYBOARD_COMMAND, g_param_spec_string("keyboard-command", NULL, NULL, NULL, G_PARAM_READWRITE));
+
+	g_object_class_install_property(object_class, PROP_MONITOR,
+		g_param_spec_int("monitor", "Xinerama monitor", "The monitor (in terms of Xinerama) which the window is on", 0, G_MAXINT,
+			0, G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
```

Comment:
Signals probably have value in being broken up but the install_property calls, dunno. Especially when one like is almost entirely just the function name.